src/configuration.h

```diff
@@ -152,10 +148,16 @@ class MetadataAgentConfiguration {
   }
 
  private:
-  friend class MetadataAgentConfigurationTest;
+  friend int ::main(int ac, char** av);
```

Comment:
Let's add a comment on why it's a `friend`. You also shouldn't need the parameter names for a `friend` declaration. So, e.g.:
```cpp
  friend int ::main(int, char**);  // Calls ParseArguments.
```
WDYT?